src/components/weather/weather.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useState } from 'react'
+
+import { getDayFromUnixTime } from '../helper'
+import Icon from '../icon/icon'
+import PopoverMenu from '../menu/popoverMenu'
+import type { WeatherProps } from '../types'
+
+/** The `Weather` component provides a detailed daily weather forecast for a given location, displaying the current temperature, weather description, today's date with high and low temperatures, and a card for each day's forecast showing the day of the week, weather icon, and high and low temperatures, all while attributing the weather data provider.
+
+Note: The first item in the `Weather` array is presumed to be the current day's weather.
+ */
+export default function Weather(props: WeatherProps) {
+  const [units, setUnits] = useState(props.units)
+
+  function getTempSymbol(units: string) {
+    switch (units) {
+      case 'imperial':
+        return 'F'
+      case 'metric':
+        return 'C'
+    }
+  }
+
+  function formatTemperature(temp: number) {
+    if (props.units === 'metric' && units === 'imperial') {
+      // celsius to fahrenheit -> °F = °C * (9/5) + 32
+      // eslint-disable-next-line no-magic-numbers
+      return Math.round(temp * (9 / 5) + 32)
+    }
+    if (props.units === 'imperial' && units === 'metric') {
+      // fahrenheit to celsius -> °C = °F * (5/9) - 32
+      // eslint-disable-next-line no-magic-numbers
+      return Math.round(temp * (5 / 9) - 32)
+    }
+
+    return Math.round(temp)
+  }
+
+  const millisecondsConversion = 1000
+  const dateStringFromTimestamp = new Date(
+    props.weather[0].timestamp * millisecondsConversion
+  )
+  const day = getDayFromUnixTime(props.weather[0].timestamp).shortName
+  const date = dateStringFromTimestamp.getDate()
+  const lastTwoDigits = 2
+  const year = dateStringFromTimestamp
+    .getFullYear()
+    .toString()
+    .slice(lastTwoDigits)
+
+  const formattedDateToday = `${day} ${date}/${year} TODAY`
+  const formattedWeatherToday = `HIGH: ${formatTemperature(props.weather[0].temp.high)} / LOW: ${formatTemperature(props.weather[0].temp.high)}`
+
+  return (
+    <Card variant="outlined" className="rustic-weather">
+      <PopoverMenu
+        ariaLabel="Temperature units"
+        menuItems={[
+          {
+            label: 'Celsius',
+            endDecorator: '°C',
+            onClick: () => setUnits('metric'),
+          },
+          {
+            label: 'Fahrenheit',
+            endDecorator: '°F',
+            onClick: () => setUnits('imperial'),
+          },
+        ]}
+      />
+      <Stack alignItems="center">
+        <Typography variant="h4" component="span" data-cy="location">
+          {props.location}
+        </Typography>
+        <Stack
+          direction="row"
+          alignItems="center"
+          spacing={1}
+          className="rustic-current-weather-description"
+          data-cy="current-weather-description"
+        >
+          <img src={props.weather[0].weatherIcon.icon} alt="" />
```

Comment:
Should we add `alt` attribute? I think it could either be `weather icon` or `${props.weather[0].weatherIcon.description} icon`

Reply:
I was thinking for this image it's not needed since the description of what it's representing immediately follows. In this case I believe this would just be considerer a decorative image so an alt tag wouldn't be needed.

ref: https://www.w3.org/WAI/tutorials/images/decorative/